views/default/elgg/Ajax.js

```diff
@@ -0,0 +1,205 @@
+				throw new Error('options must be a plain with key "url"');
+			}
+
+			// ease hook filtering by making these keys always available
+			options.data = options.data || {};
+			options.dataType = 'json';
+			if (!options.method) {
+				options.method = 'GET';
+			}
+
+			// copy of original options
+			orig_options = $.extend({}, options);
+
+			params = {
+				options: options
+			};
+			if (hook_type) {
+				options.data = elgg.trigger_hook(Ajax.REQUEST_DATA_HOOK, hook_type, params, options.data);
+			}
+
+			if ($.isArray(options.success)) {
+				throw new Error('The array form of options.success is not supported');
+			}
+
+			if (elgg.isFunction(options.success)) {
+				options.success = function (data) {
+					data = unwrap_data(data);
+					orig_options.success(data);
+				};
+			}
+
+			if (use_spinner) {
+				options.beforeSend = function () {
+					orig_options.beforeSend && orig_options.beforeSend();
+					spinner.start();
+				};
+				options.complete = function () {
+					spinner.stop();
+					orig_options.complete && orig_options.complete();
+				};
+			}
+
+			if (!options.error) {
+				options.error = elgg.ajax.handleAjaxError;
+			}
+
+			options.url = elgg.normalize_url(options.url);
+			options.headers = {
+				'X-Elgg-Ajax-API': '2'
+			};
+
+			return $.ajax(options).then(unwrap_data);
+		}
+
+		/**
+		 * Fetch content from a page handler
+		 *
+		 * See fetch() for more options.
+		 *
+		 * @param {String} path    URL path. e.g. "foo/bar"
+		 * @param {Object} options {@link jQuery#ajax}. See fetch()
+		 * @returns {jqXHR}
+		 */
+		this.fetchPath = function (path, options) {
+			elgg.assertTypeOf('string', path);
+
+			options = options || {};
+			options.url = path;
+
+			return fetch(options, 'path:' + path.replace(/\/$/, ''));
+		};
+
+		/**
+		 * Fetch content from a view registered for Ajax
+		 *
+		 * See fetch() for more options.
+		 *
+		 * @param {String} view    View name to fetch
+		 * @param {Object} options {@link jQuery#ajax}. See fetch()
+		 * @returns {jqXHR}
+		 */
+		this.fetchView = function (view, options) {
+			elgg.assertTypeOf('string', view);
+
+			options = options || {};
+			options.url = 'ajax/view/' + view;
+
+			return fetch(options, 'view:' + view);
+		};
+
+		/**
+		 * Fetch a form rendered on the server
+		 *
+		 * See fetch() for more options.
+		 *
+		 * @param {String} action  Action name
+		 * @param {Object} options {@link jQuery#ajax}. See fetch()
+		 * @returns {jqXHR}
+		 */
+		this.fetchForm = function (action, options) {
+			elgg.assertTypeOf('string', action);
+
+			options = options || {};
+			options.url = 'ajax/form/' + action;
+
+			return fetch(options, 'form:' + action);
+		};
+
+		/**
+		 * Perform an action
+		 *
+		 * See fetch() for more options.
+		 *
+		 * @param {String} action  Action name
+		 * @param {Object} options jQuery.ajax options. See fetch()
+		 * @returns {jqXHR}
+		 */
+		this.performAction = function (action, options) {
```

Comment:
Nitpicking, but I'd have slight preference on just `action()` instead of `performAction()`.